dagql/server.go

```diff
@@ -317,19 +320,27 @@ func (s *Server) Schema() *ast.Schema { // TODO: change this to be updated whene
 	defer s.installLock.Unlock()
 	// TODO track when the schema changes, cache until it changes again
 	queryType := s.Root().Type().Name()
-	schema := &ast.Schema{}
+	schema := &ast.Schema{
+		Types:         make(map[string]*ast.Definition),
+		PossibleTypes: make(map[string][]*ast.Definition),
+	}
 	for _, t := range s.objects { // TODO stable order
 		def := definition(ast.Object, t, s.View)
 		if def.Name == queryType {
 			schema.Query = def
 		}
 		schema.AddTypes(def)
+		schema.AddPossibleType(def.Name, def)
```

Comment:
Is this right? I think "possible types" is meant to be a mapping from unions and interfaces to their concrete implementations, whereas this seems to imply every type has itself as a possible type

Reply:
Yeah, I don't know actually :thinking: 

But this validation rule implies that it is meant to be set like this? https://github.com/vektah/gqlparser/blob/e21b122b4e0ebb42ad98fe4ecf738bf380e487a3/validator/rules/possible_fragment_spreads.go#L36-L36

Reply:
Does it fail without it?

The linked is regarding fragment spreads which should only be relevant to type switching on interfaces/enums, but it's for sure weird that it seems to pass the spreaded (concrete) type to `GetPossibleTypes` considering its own godoc says it's only for interfaces/enums:

```go
// GetPossibleTypes will enumerate all the definitions for a given interface or union
func (s *Schema) GetPossibleTypes(def *Definition) []*Definition {
	return s.PossibleTypes[def.Name]
}
```

Either way if it doesn't fail I would just skip it - we don't use interfaces anyway (and I don't think we use unions either)

Reply:
It does fail, which was the reason I actually caught that linting wasn't ever enabled in the first place:

```
TransportQueryError: Error while fetching schema: {'message': 'Fragment 
"TypeRef" cannot be spread here as objects of type "__Type" can never be of type
"__Type".', 'locations': [{'line': 63, 'column': 8}], 'extensions': {'code': 
'GRAPHQL_VALIDATION_FAILED'}}
```

From https://github.com/vektah/gqlparser/blob/e21b122b4e0ebb42ad98fe4ecf738bf380e487a3/validator/rules/possible_fragment_spreads.go#L58-L68

I think potentially maybe this is just... wrong? Might submit a patch upstream.

Reply:
Ah, this is why it works - as part of loading the schema, it adds all `Type`s to `PossibleType`s :thinking: https://github.com/vektah/gqlparser/blob/28ef20ae1564c8a8ac8a0305711e0122fe937e9a/validator/schema.go#L74-L74

So we should do the same thing here.

Reply:
Alright - seems weird, but I'm convinced it's at least consistent 😛